src/math/CMakeLists.txt

```diff
@@ -5,31 +5,41 @@ if(BUILD_LIBRARY)
 	return()
 endif()
 
+is_zephyr(it_is_zephyr)
+
 add_local_sources(sof numbers.c)
 
-if(CONFIG_CORDIC_FIXED)
+if(CONFIG_CORDIC_FIXED OR it_is_zephyr)
         add_local_sources(sof trig.c)
 endif()
 
 add_local_sources_ifdef(CONFIG_SQRT_FIXED sof sqrt_int16.c)
 
 add_local_sources_ifdef(CONFIG_MATH_EXP sof exp_fcn.c exp_fcn_hifi.c)
 
-if(CONFIG_MATH_DECIBELS)
+if(CONFIG_MATH_DECIBELS OR it_is_zephyr)
         add_local_sources(sof decibels.c)
 endif()
 
+# Never used with Zephyr.
+if(NOT it_is_zephyr)
 add_local_sources_ifdef(CONFIG_NATURAL_LOGARITHM_FIXED sof log_e.c)
 
 add_local_sources_ifdef(CONFIG_COMMON_LOGARITHM_FIXED sof log_10.c)
 
 add_local_sources_ifdef(CONFIG_POWER_FIXED sof power.c)
 
 add_local_sources_ifdef(CONFIG_BINARY_LOGARITHM_FIXED sof base2log.c)
+endif()
 
+if(it_is_zephyr)
+add_local_sources_ifdef(CONFIG_COMP_FIR sof fir_generic.c fir_hifi2ep.c fir_hifi3.c)
```

Comment:
why not CONFIG_MATH_FIR?  TDFB will select  MATH_FIR too.

Reply:
Because it was `CONFIG_MATH_FIR` in Zephyr, please pay attention to the previous code.

Please read the previous discussion: the purpose of this PR is to carefully NOT change the CONFIGuration.

If you want to fix the configuration then please submit a separate PR and I will adjust this one.

Reply:
I think it is a mistake in zephyr/Cmakelists.txt, the fir lib files should be like 
```
zephyr_library_sources_ifdef(CONFIG_MATH_FIR
	${SOF_MATH_PATH}/fir_generic.c
	${SOF_MATH_PATH}/fir_hifi2ep.c
	${SOF_MATH_PATH}/fir_hifi3.c
)
```

Reply:
here is the fix https://github.com/thesofproject/sof/pull/8581